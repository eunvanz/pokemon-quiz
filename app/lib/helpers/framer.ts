import { Variants } from 'framer-motion'

export const motionVariants: Variants = {
  vibe: {
    translateX: [0, -4, 0, 4, 0],
    transition: {
      repeat: 2,
      duration: 0.1,
    },
  },
}
