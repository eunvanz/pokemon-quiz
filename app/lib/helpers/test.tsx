import CommonProvider from '@/app/components/common-provider'
import { PropsWithChildren } from 'react'
import api from '../api'

export interface TestProviderProps {
  api?: Partial<typeof api>
}

export const TestProvider: React.FC<PropsWithChildren<TestProviderProps>> = ({
  children,
  api = {},
}) => <CommonProvider api={api}>{children}</CommonProvider>
