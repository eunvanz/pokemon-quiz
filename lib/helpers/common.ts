export const checkIsSSR = () => {
  return typeof window === 'undefined'
}
