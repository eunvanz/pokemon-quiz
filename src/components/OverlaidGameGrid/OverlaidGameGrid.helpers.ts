export const convertStackedMonImagesToArray: (
  stackedMonImages: string[][],
) => string[] = (stackedMonImages) => {
  const colSize = stackedMonImages.length;
  const result: string[] = [];

  stackedMonImages.forEach((rows, colIndex) => {
    rows.forEach((item, rowIndex) => {
      const index = rowIndex * colSize + colIndex;
      result[index] = item;
    });
  });
  return result;
};

export const getPositionFrom2DArray: (
  colSize: number,
  index: number,
) => { x: number; y: number } = (colSize, index) => {
  const x = colSize - (index % colSize) - 1;
  const y = Math.floor(index / colSize);
  return { x, y };
};

export const getStackedSizeFromStackedMonImages: (
  stackedMonImages: string[][],
  colIndex: number,
) => number = (stackedMonImages, colIndex) => {
  return stackedMonImages[colIndex].length;
};
