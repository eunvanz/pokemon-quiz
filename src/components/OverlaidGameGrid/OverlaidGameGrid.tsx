import GameGrid from "../GameGrid";

export interface OverlaidGameGridProps {
  currentMonImage: string;
  duration: number;
  stackedMonImages: string[][];
}

const OverlaidGameGrid: React.FC<OverlaidGameGridProps> = ({
  currentMonImage,
  duration,
  stackedMonImages,
}) => {
  return (
    <>
      <GameGrid gridItemSize={6} width={300} />
    </>
  );
};

export default OverlaidGameGrid;
