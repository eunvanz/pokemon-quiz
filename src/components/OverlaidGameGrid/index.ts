export { default } from "./OverlaidGameGrid";

export type { OverlaidGameGridProps } from "./OverlaidGameGrid";
