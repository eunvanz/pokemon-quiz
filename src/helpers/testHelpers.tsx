import { Story } from "@storybook/react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClientProvider } from "react-query";
import api from "~/api";
import { ApiProvider } from "~/api/apiContext";
import { queryClient } from "./reactQuery";

/**
 * render storybook component in test code
 * @param StoryComponent
 * @param props
 * @returns
 */
export const renderStory = <T,>(
  StoryComponent: Story<Partial<T>>,
  props?: Partial<T>,
  renderOptions?: RenderOptions,
) => {
  return render(<StoryComponent {...props} />, renderOptions);
};

export interface TestProviderProps {
  api: Partial<typeof api>;
}

export const TestProvider: React.FC<TestProviderProps> = ({ children, api }) => (
  <ApiProvider api={api}>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </ApiProvider>
);
