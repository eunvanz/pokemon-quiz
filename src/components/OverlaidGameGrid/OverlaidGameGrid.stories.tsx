import type { ComponentStory, ComponentMeta } from "@storybook/react";
import { createStoryComponent } from "~/helpers/storybookHelpers";
import OverlaidGameGrid from "./OverlaidGameGrid";

export default {
  title: "components/OverlaidGameGrid",
  component: OverlaidGameGrid,
  args: {
    currentMonImage:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/132.png",
    currentColumn: 3,
    duration: 20,
  },
} as ComponentMeta<typeof OverlaidGameGrid>;

const Template: ComponentStory<typeof OverlaidGameGrid> = (args) => (
  <OverlaidGameGrid {...args} />
);

export const Default = createStoryComponent(Template);
