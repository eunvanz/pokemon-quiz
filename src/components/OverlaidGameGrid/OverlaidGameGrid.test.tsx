import { ComponentProps } from "react";
import { composeStories } from "@storybook/testing-react";
import { renderStory } from "~/helpers/testHelpers";
import OverlaidGameGrid from "./OverlaidGameGrid";
import * as stories from "./OverlaidGameGrid.stories";

const { Default } = composeStories(stories);

describe("OverlaidGameGrid", () => {
  const setup = (props?: ComponentProps<typeof OverlaidGameGrid>) =>
    renderStory(Default, props);

  it("matches with previous snapshot.", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
