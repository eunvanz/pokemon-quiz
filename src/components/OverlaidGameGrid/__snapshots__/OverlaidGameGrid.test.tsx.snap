// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlaidGameGrid matches with previous snapshot. 1`] = `
<div>
  <div
    class="css-ywlqmn-OverlaidGameGrid"
  >
    <div
      class="css-i5fm1e-OverlaidGameGrid"
    >
      <div
        class="css-fqtnk-OverlaidGameGrid"
      />
      <div
        class="css-fqtnk-OverlaidGameGrid"
      />
      <div
        class="css-fqtnk-OverlaidGameGrid"
      />
      <div
        class="css-fqtnk-OverlaidGameGrid"
      >
        <div>
          <img
            alt="mon image"
            class="css-10yqwrw-MonImg"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/132.png"
            style="transform: translateY(0px) scale(0) translateZ(0);"
            width="50"
          />
        </div>
      </div>
      <div
        class="css-fqtnk-OverlaidGameGrid"
      />
      <div
        class="css-fqtnk-OverlaidGameGrid"
      />
    </div>
    <div
      class="css-17somsz-OverlaidGameGrid"
    >
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/11.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/10.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/9.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/7.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/6.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/5.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/3.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/4.png"
          width="50"
        />
      </div>
      <div
        class="css-1w13e65-OverlaidGameGrid"
      />
      <div
        class="css-1w13e65-OverlaidGameGrid"
      >
        <img
          alt="mon image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/8.png"
          width="50"
        />
      </div>
    </div>
    <div
      class="css-1nk0c6d-GameGrid"
      data-testid="game-grid-container"
    >
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
      <div
        class="css-1mc37or-GameGrid"
      />
    </div>
  </div>
</div>
`;
